packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/lambda/call-aws-service-cross-region.ts

```diff
@@ -0,0 +1,184 @@
+
+  /**
+   * Parameters for the API action call.
+   *
+   * Use PascalCase for the parameter names.
+   *
+   * @default - no parameters
+   */
+  readonly parameters?: { [key: string]: any };
+
+  /**
+   * The resources for the IAM statement that will be added to the Lambda
+   * function role's policy to allow the state machine to make the API call.
+   *
+   * By default the action for this IAM statement will be `service:action`.
+   */
+  readonly iamResources: string[];
+
+  /**
+   * The action for the IAM statement that will be added to the Lambda
+   * function role's policy to allow the state machine to make the API call.
+   *
+   * Use in the case where the IAM action name does not match with the
+   * API service/action name, e.g. `s3:ListBuckets` requires `s3:ListAllMyBuckets`.
+   *
+   * @default - service:action
+   */
+  readonly iamAction?: string;
+
+  /**
+   * Additional IAM statements that will be added to the state machine
+   * role's policy.
+   *
+   * Use in the case where the call requires more than a single statement to
+   * be executed, e.g. `rekognition:detectLabels` requires also S3 permissions
+   * to read the object on which it must act.
+   *
+   * @default - no additional statements are added
+   */
+  readonly additionalIamStatements?: iam.PolicyStatement[];
+
+  /**
+   * The AWS region to call this AWS API for.
+   * @example 'us-east-1'
+   */
+  readonly region: string;
+
+  /**
+   * The AWS API endpoint.
+   *
+   * @default Do not override API endpoint.
+   */
+  readonly endpoint?: string;
+
+  /**
+   * Whether to retry on the backend Lambda service exceptions.
+   *
+   * This handles `Lambda.ServiceException`, `Lambda.AWSLambdaException`,
+   * `Lambda.SdkClientException`, and `Lambda.ClientExecutionTimeoutException`
+   * with an interval of 2 seconds, a back-off rate
+   * of 2 and 6 maximum attempts.
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/dg/bp-lambda-serviceexception.html
+   *
+   * @default true
+   */
+  readonly retryOnServiceExceptions?: boolean;
+}
+
+/**
+ * A Step Functions task to call an AWS service API across regions.
+ *
+ * This task creates a Lambda function to call cross-region AWS API and invokes it.
+ */
+export class CallAwsServiceCrossRegion extends sfn.TaskStateBase {
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+  protected readonly lambdaFunction: IFunction;
+
+  constructor(scope: Construct, id: string, private readonly props: CallAwsServiceCrossRegionProps) {
+    super(scope, id, props);
+
+    if (props.integrationPattern === sfn.IntegrationPattern.RUN_JOB) {
+      throw new Error('The RUN_JOB integration pattern is not supported for CallAwsService');
+    }
+    if (!Token.isUnresolved(props.action) && !props.action.startsWith(props.action[0]?.toLowerCase())) {
+      throw new Error(`action must be camelCase, got: ${props.action}`);
+    }
```

Comment:
I know you mentioned that it's similar to `CallAwsService` where no validation/normalization is done. I would still recommend using the `normalizeServiceName` and `normalizeActionName`. If users input something like `getobject` we wouldn't catch it. While they would eventually know from the logs/execution results, why not normalize it ahead?

Reply:
I think your example `getobject` is much more difficult to catch. Afaik there has been no complaint about `CallAwsService` only supporting camelCase, so I don't think it's worth supporting other formats in this construct either, at least at this moment.